### PR TITLE
Add support for axios-concerrency

### PR DESCRIPTION
**Pull request by powfix**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/axios-concurrency/index.d.ts
1. types/axios-concurrency/axios-concurrency-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 16d4009a51f61b80aafa84b4092a066de4705ba0
</p>

File: types/axios-concurrency/axios-concurrency-tests.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+import { ConcurrencyManager, ConcurrencyManagerInstance } from "axios-concurrency";
+
+const axiosInstance = axios.create({});
+
```
Comment (RyanCavanaugh) [inline types/axios-concurrency/axios-concurrency-tests.ts:5]: There should really be at least one *passing* test here

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/axios-concurrency/axios-concurrency-tests.ts (hunk 1 above)
Comment (powfix) [reply types/axios-concurrency/axios-concurrency-tests.ts:5]: Could the below code be a passing test?

```
const manager: ConcurrencyManagerInstance = ConcurrencyManager(axiosInstance, 1);

// $ExpectType void
manager.detach();
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@powfix thanks!

**Comment by powfix** [comment]
> @powfix: Everything looks good here. I am ready to merge this PR (at [1c80269](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/1c80269d8a847b728af8a5502fa759ee9586fe09)) on your behalf whenever you think it's ready.
> 
> If you'd like that to happen, please post a comment saying:
> 
> > Ready to merge
> 
> and I'll merge this PR almost instantly. Thanks for helping out! ❤️

**Comment by powfix** [comment]
Ready to merge

**Pull request merged**